### PR TITLE
Add maintenance page

### LINKED ISSUE #29: Add Maintenance Mode
Since our site is currently a work in progress, it would be great to have a simple way to enable "maintenance mode", ideally through environment variables, which will load a simple page telling people that the site is currently under development.

I haven't dug into it yet, but [this Vercel template](https://vercel.com/templates/next.js/maintenance-page) might be useful.

Once we have maintenance mode set up, we can put the production site into maintenance mode until we're ready to welcome users!

### PR DESCRIPTION
**Pull request by aberonni**
- Add maintenance page
- Add middleware that redirects all requests to maintenance page when the edge config says so
- Move "Header and footer" layout to (root)/layout.tsx, and subsequently move all existing pages into that page group (https://nextjs.org/docs/app/building-your-application/routing/route-groups)

Closes #29

### CODE REVIEW

**Comment by aberonni** [comment]
You should be able to test this by visiting the Vercel preview for this PR and then toggling the edge config in the Vercel dashboard. The website should switch between states. Let me know what you think!

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  ],
+};
+
+export async function middleware(req: NextRequest) {
+  let isInMaintenanceMode = false;
+
+  try {
+    isInMaintenanceMode =
+      // Don't show the maintenance page in development
+      process.env.NODE_ENV !== "development" &&
```
Comment (rlmead) [inline src/middleware.ts:30]: It looks to me like all Vercel deployments use the same store, so I was wondering about how we keep maintenance mode off for the dev deployment for testing (while leaving prod in maintenance mode). Looks like it's this, yeah? Could you add `NODE_ENV` to .env.sample? I'll add it on Vercel, with values `development` and `production`. Thank ya!

**Comment by rlmead** [comment]
> You should be able to test this by visiting the Vercel preview for this PR and then toggling the edge config in the Vercel dashboard. The website should switch between states. Let me know what you think!

@aberonni It works perfectly!! Just one minor comment to address before merging, but everything else looks great to me 😄 Thank you!

File: src/middleware.ts (hunk 1 above)
Comment (aberonni) [reply src/middleware.ts:30]: Oh, there's no need to add `NODE_ENV` manually anywhere. By default it is always 'development' and Next/React add `NODE_ENV=production` to the build without any manual intervention. It's a common convention and doesn't require any extra work to do it's magic. More info: https://nodejs.org/en/learn/getting-started/nodejs-the-difference-between-development-and-production

Comment (aberonni) [reply src/middleware.ts:30]: And of course

> Looks like it's this, yeah?

Yes indeed! That's my thinking, but happy to go for a different solution if you like.

Comment (rlmead) [reply src/middleware.ts:30]: Amazing, I didn't know all that! Thanks for the explanation 🧠

**Review by rlmead — APPROVED** [review]

**Comment by rlmead** [comment]
@aberonni FYI these are the merging conventions that I follow:
* squash merge into `dev`
* merge commit into `main`

Let me know if you know of any reason to do things differently! I realize there's only one commit here but I'm gonna go ahead and squash merge out of habit and superstition 😆

**Comment by aberonni** [comment]
> @aberonni FYI these are the merging conventions that I follow:
> 
> * squash merge into `dev`
> * merge commit into `main`
> 
> Let me know if you know of any reason to do things differently! I realize there's only one commit here but I'm gonna go ahead and squash merge out of habit and superstition 😆

I've pretty much always just done a simple merge, but with an enforcement on branches to be rebased before merging. I don't know the specifics of why I've ended up doing that - it's something a wiser programmer than me passed on to me eons ago :D

I have no opinion or preference so your flow sounds good to me ✌️

**Pull request merged**